js/web/lib/wasm/jsep/webgpu/ops/3rd-party/conv2d_mm_webgpu.ts

```diff
@@ -143,7 +143,15 @@ const conv2dCommonSnippet = (
     }
     return ${typeSnippet(innerElementSizeX, dataType)}(0.0);`;
 
-  const sampleW = `${getWSnippet(innerElementSizeW)}`;
+  const sampleW =
+    fitInner && fitBOuter
+      ? getWSnippet(innerElementSizeW)
+      : `
+    let col = colIn * ${innerElementSizeW};
+    if (row < uniforms.dim_inner && col < uniforms.dim_b_outer) {
+      ${getWSnippet(innerElementSizeW)}
+    }
+    return ${typeSnippet(innerElementSizeW, dataType)}(0.0);`;
```

Comment:
This is for channel last case. The channel first is missing.
```suggestion
  const sampleW = isChannelsLast ?
  fitInner && fitBOuter
    ? getWSnippet(innerElementSizeW)
    : `
  let col = colIn * ${innerElementSizeW};
  if (row < uniforms.dim_inner && col < uniforms.dim_b_outer) {
    ${getWSnippet(innerElementSizeW)}
  }
  return ${typeSnippet(innerElementSizeW, dataType)}(0.0);`
  : `
  let col = colIn * ${innerElementSizeW};
  if (row < uniforms.dim_inner && col < uniforms.dim_a_outer) {
    ${getWSnippet(innerElementSizeW)}
  }
  return ${typeSnippet(innerElementSizeW, dataType)}(0.0);`;
```